docs/assets/images/vip_3_read_path.drawio.xml

```diff
@@ -0,0 +1,212 @@
+<mxfile host="app.diagrams.net" modified="2023-08-23T13:47:02.064Z" agent="Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.6 Safari/605.1.15" etag="jDuoTWvUTu5oXmrRuByY" version="21.6.9" type="device">
```

Comment:
Confused on why there is two paths from FC -> RustServer via gRPC Handler? 

Why do we need the gPRC Request Handler - JNI call -> Rust Server while we have FC directly reaching out to RUST server which embeds a gRPC Request handler ?

Reply:
This is a transitory stage, where we would support both the legacy and new paths at the same time. The end state would be that there's no Java code left to handle read requests (though that is probably far out...).

Reply:
Ahh I see, while its getting tested. Thanks for clarification.